scipy/spatial/transform/_rotation.pyx

```diff
@@ -1417,7 +1535,48 @@ cdef class Rotation:
             return np.asarray(rotvec)
 
     @cython.embedsignature(True)
-    def as_euler(self, seq, degrees=False):
+    def _compute_euler(self, seq, degrees, algorithm):
+        # Prepare axis sequence to call Euler angles conversion algorithm.
+        
+        if len(seq) != 3:
+            raise ValueError("Expected 3 axes, got {}.".format(seq))
+
+        intrinsic = (re.match(r'^[XYZ]{1,3}$', seq) is not None)
+        extrinsic = (re.match(r'^[xyz]{1,3}$', seq) is not None)
+        if not (intrinsic or extrinsic):
+            raise ValueError("Expected axes from `seq` to be from "
+                             "['x', 'y', 'z'] or ['X', 'Y', 'Z'], "
+                             "got {}".format(seq))
+
+        if any(seq[i] == seq[i+1] for i in range(2)):
+            raise ValueError("Expected consecutive axes to be different, "
+                             "got {}".format(seq))
+
+        seq = seq.lower()
+            
+        if algorithm == 'from_matrix':
+            matrix = self.as_matrix()
+            if matrix.ndim == 2:
+                matrix = matrix[None, :, :]
+            angles = np.asarray(_compute_euler_from_matrix(
+                matrix, seq.encode(), extrinsic))
+        elif algorithm == 'from_quat':
+            quat = self.as_quat()
+            if quat.ndim == 1:
+                quat = quat[None, :]
+            angles = np.asarray(_compute_euler_from_quat(
+                    quat, seq.encode(), extrinsic))
+        else:
+            raise ValueError("algorithm can be 'from_quat' or 'from_matrix'"
```

Comment:
I guess `assert False` would be more idiomatic here. Meaning that this case is caused by incorrect internal code usage. Or in other word: execution should never get to this point.

Reply:
I see what you mean. Should I just delete this check then? I thought of leaving it as a practical error message in case someone decides to so some kind of internal changes. This way they know how to call them exactly.

What should I replace this with exactly?

Reply:
You can keep it, or replace with literally `assert False`. The big reason for that is to avoid spending time on writing exceptions in such places.

Reply:
Alright, done!